packages/client/src/connection.ts

```diff
@@ -142,9 +150,16 @@ function addDefaults(options: ConnectionOptions): ConnectionOptionsWithDefaults
  * - Convert {@link ConnectionOptions.tls} to {@link grpc.ChannelCredentials}
  * - Add the grpc.ssl_target_name_override GRPC {@link ConnectionOptions.channelArgs | channel arg}
  * - Add default port to address if port not specified
+ * - Set `Authorization` header based on API key (and not already set explicitly)
  */
 function normalizeGRPCConfig(options?: ConnectionOptions): ConnectionOptions {
   const { tls: tlsFromConfig, credentials, callCredentials, ...rest } = options || {};
```

Comment:
```suggestion
  const { tls: tlsFromConfig, credentials, callCredentials, metadata = {}, ...rest } = options || {};
```

... and then, replace `rest.metadata` everywhere below. No more need to check that metadata isn't empty.

Reply:
Yeah, forgot to mention that you would have to add `metadata` in object spread later on. That's still more JS-idiomatic.